### PR TITLE
Do not use = {nullptr} initialization.

### PR DESCRIPTION
**Pull request by sandreenko**
It works bad on desktop, that uses old C++ compiler (Version 18.10.40116.12).

For such code in flowgraph.cpp : fgInvokeInlineeCompiler 
```
// This is the InlineInfo struct representing a method to be inlined.
InlineInfo inlineInfo = {nullptr};
unsigned char *p = (unsigned char *)(void*)&inlineInfo;
int i;
for (i = 0; i < sizeof InlineInfo; i++)
{
  printf("%02x ", p[i]);
}
```
It prints 

```
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 20 00 00 00 03 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 02 00 00 00 00 00 00 00 00 00 00 00 69 00 00 00 00 00 00 00 
	00 00 00 00 18 d3 7c 5a 69 00 00 00 00 00 00 00 69 00 00 00 00 00 00 00 00 00 00 00 e0 
	d2 7c 5a 69 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 69 00 00 00 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 
	00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 09 00 00 00 
	00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 
	00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc 
	cc cc 0b 00 00 00 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 06 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 
	cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 
	00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 f6 7f 00 00 00 00 00 00 00 00 
	00 00 00 08 2f 2e f9 7f 00 00 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 10 70 bb 
	2c f9 7f 00 00 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 
	00 00 00 00 7b 27 00 00 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 f8 
	7f 00 00 00 00 00 00 00 00 00 00 06 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 
	00 cc cc cc cc cc cc cc cc 00 00 00 00 09 00 00 00 00 00 00 00 00 00 00 00 cc cc cc cc 
	cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 
	00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc 0a 00 00 00 00 00 00 00 cc cc 
	cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 
	00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 
	cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc 
	cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 7f 14 94 df f8 7f 00 00 00 00 
	00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc 
	cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 
	00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc 
	cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 cc cc cc cc cc cc 
	cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 6c d5 7c 5a 69 00 00 00 00 00 00 
	00 69 00 00 00 00 00 00 00 00 00 00 00 70 eb bc 22 c9 01 00 00 00 00 00 00 69 00 00 00 
	00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 09 00 00 00 00 00 00 00 00 
	00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 f8 7f 00 00 00 00 00 00 00 00 00 00 00 cc 
	cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 fe ff ff ff ff ff ff 
	ff 00 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 bf 87 a0 df f8 7f 00 00 00 00 00 00 
	c9 01 00 00 00 00 00 00 00 00 00 00 cc cc cc cc cc cc cc cc 00 00 00 00 cc cc cc cc 00 
	00 00 00 00 00 00 00 2d 5f 91 df f8 7f 00 00 00 00 00 00 c9 01 00 00 00 00 00 00 00 00 
	00 00 b0 b9 7f 24 c9 01 00 00 00 00 00 00 f8 7f 00 00 00 00 00 00 00 00 00 00 60 00 00 
	00 00 00 00 00 00 00 00 00 ff ff ff ff 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
	00 00 00 00 00 00 00 00 00 00 00 b0 b9 7f 24 c9 01 00 00 00 00 00 00 f8 7f 00 00 00 00 
	00 00 00 00 00 00 5b 00 00 00 cc cc cc cc 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 ea ba 1e 3a 70 39 00 00 00 00 00 00 f8 7f 00 00 00 00 00 00 00 00 00 00 40 d7 7c 5a 
	69 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 88 68 7f 24 c9 01 00 00 00 
	00 00 00 f8 7f 00 00 00 00 00 00 00 00 00 00 5b 00 00 00 00 00 00 00 00 00 00 00 c9 01 
	00 00 00 00 00 00 00 00 00 00 01 a1 ab df 02 02 00 00 00 00 00 00 f8 7f 00 00 00 00 00 
	00 00 00 00 00 6a 39 90 df f8 7f 00 00 00 00 00 00 c9 01 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 f8 7f 00 00 00 00 00 00 00 00 00 00 6a 39 90 df f8
	00 00 69 00 00 00 00 00 00 00 00 00 00 00 c7 2e 90 df f8 7f 00 00 00 00 00 00 70 39 00 
	00 00 00 00 00 00 00 00 00 70 64 7f 24 c9 01 00 00 00 00 00 00 69 00 00 00 00 00 00 00 
	00 00 00 00 5f 82 94 df f8 7f 00 00 00 00 00 00 c9 01 00 00 00 00 00 00 00 00 00 00 c0 
	12 94 df f8 7f 00 00 00 00 00 00 f8 7f 00 00 00 00 00 00 00 00 00 00 00 cc cc cc cc cc 
	cc cc 00 00 00 00 02 02 00 00 00 00 00 00 00 00 00 00 fe ff ff ff ff ff ff ff 00 00 00 
	00 cc cc cc cc 00 00 00 00 00 00 00 00 bf 87 a0 df f8 7f 00 00 00 00 00 00 00 00 00 00 
	00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
	00 00 00
```
and generates such asm (checked amd64):
```
	InlineInfo inlineInfo = {nullptr};
00007FF8DF9358E7  xor         edx,edx  
00007FF8DF9358E9  mov         qword ptr [rbp-80h],r13  
00007FF8DF9358ED  lea         rcx,[rbp-78h]  
00007FF8DF9358F1  lea         r8d,[rdx+38h]  
00007FF8DF9358F5  call        memset (07FF8DFA94720h)  
00007FF8DF9358FA  xor         edx,edx  
00007FF8DF9358FC  lea         rcx,[rbp-40h]  
00007FF8DF935900  mov         r8d,318h  
00007FF8DF935906  call        memset (07FF8DFA94720h)  
00007FF8DF93590B  xor         edx,edx  
00007FF8DF93590D  lea         rcx,[rbp+2D8h]  
00007FF8DF935914  mov         r8d,80h  
00007FF8DF93591A  call        memset (07FF8DFA94720h)  
00007FF8DF93591F  mov         edx,18h  
00007FF8DF935924  lea         r9,[InlLclVarInfo::InlLclVarInfo (07FF8DF920C00h)+100000000h]  
00007FF8DF93592B  lea         rcx,[rbp+358h]  
00007FF8DF935932  lea         r8d,[rdx+29h]  
00007FF8DF935936  call        `vector constructor iterator' (07FF8DF921180h)+100000000h  
00007FF8DF93593B  xor         edx,edx  
00007FF8DF93593D  lea         rcx,[rbp+970h]  
00007FF8DF935944  lea         r8d,[rdx+28h]  
00007FF8DF935948  call        memset (07FF8DFA94720h) 
```

It creates non-deterministic values of different flags, for example localVariable.lclHasMultipleStlocOp. It creates spmi errors, because depends on this values it asks different questions.

CoreCLR with VS2015 generates such asm code:
```
    InlineInfo inlineInfo = {nullptr};
00007FF8DDCBC427  xor         edx,edx  
00007FF8DDCBC429  lea         rcx,[inlineInfo]  
00007FF8DDCBC42E  mov         r8d,9F0h  
00007FF8DDCBC434  call        memset (07FF8DDE0FAA0h)  
00007FF8DDCBC439  mov         ecx,41h  
00007FF8DDCBC43E  lea         rax,[rbp+338h]  
00007FF8DDCBC445  xor         edi,edi  
00007FF8DDCBC447  mov         dword ptr [rax],edi  
00007FF8DDCBC449  mov         qword ptr [rax+8],rdi  
00007FF8DDCBC44D  lea         rax,[rax+18h]  
00007FF8DDCBC451  sub         rcx,1  
00007FF8DDCBC455  jne         Compiler::fgInvokeInlineeCompiler+0BBh (07FF8DDCBC447h)  
00007FF8DDCBC457  lea         r8d,[rcx+28h]  
00007FF8DDCBC45B  xor         edx,edx  
00007FF8DDCBC45D  lea         rcx,[rbp+950h]  
00007FF8DDCBC464  call        memset (07FF8DDE0FAA0h) 
```
and it works right.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @pgavlin @BruceForstall @dotnet/jit-contrib .

**Comment by sandreenko** [comment]
As I understand it didn't create test failures because we reassigned most values then, the exception was lclHasMultipleStlocOp that forbids optimizations with set value, but didn't cause conformance problems. It was added recently and it explains why spmi problems has appeared.

File: src/jit/lsra.cpp [hunk 1]
```diff
@@ -9677,10 +9677,12 @@ void LinearScan::resolveEdge(BasicBlock*      fromBlock,
 
     // What interval is this register associated with?
     // (associated with incoming reg)
-    Interval* sourceIntervals[REG_COUNT] = {nullptr};
+    Interval* sourceIntervals[REG_COUNT];
+    memset(&sourceIntervals, 0, sizeof(location));
```
Comment (mikedn) [inline src/jit/lsra.cpp]: `sizeof(location)`?!

Comment (sandreenko) [inline src/jit/lsra.cpp]: My bad, thanks!

**Comment by sandreenko** [comment]
> Does the older compiler produce similarly incorrect code gen for = {0}?
Yes.

**Pull request merged**